src/gpb_compile.erl

```diff
@@ -6331,14 +6331,10 @@ mandatory_map_item_type_sep(Opts) ->
 can_specify_map_item_presence_in_typespecs(Opts) ->
     is_target_major_version_at_least(19, Opts).
 
-type_to_typestr(MsgName, #?gpb_field{type=Type, occurrence=Occurrence},
+type_to_typestr(#?gpb_field{type=Type, occurrence=Occurrence},
                 Defs, Opts) ->
     OrUndefined = case get_mapping_and_unset_by_opts(Opts) of
-                      records                   ->
-                          case gpb:is_msg_proto3(MsgName, Defs)  of
-                              true -> "";
```

Comment:
I don't think that this actually does what we want.  Could you paste an example of the warning that you're trying to fix?